### PR TITLE
add basic vertical localization

### LINKED ISSUE #729: add basic vertical obs localization to ufo and soca
## Description
add basic obs localization with distance specified in units of depth. 

## Related to
part of the vertical localization epic

### PR DESCRIPTION
**Pull request by frolovsa**
## Description
Introduces two tests for vertical obs localization. 
- test_soca_obslocalization_vertical tests different parts of the vertical localization (different localization function, log transforms, etc)
- test_soca_obslocalizations tests sequential application of horizontal and vertical localization

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #729 

## Testing
Tested on orion using new ctets

## Dependencies
uses matching branch in ufo 
- waiting on JCSDA-internal/ufo/pull/1959

### CODE REVIEW

File: test/testinput/obslocalization_vertical.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+geometry:
```
Comment (guillaumevernieres) [inline test/testinput/obslocalization_vertical.yml:1]: It's not as slow as your other vertical loc test (~130s), but it's still in the 10s range. Could you try to use the subsampled grid? Pointing to the geometry below should do the trick.
```yaml
geometry:
    geom_grid_file: soca_gridspec.small.nc
    mom6_input_nml: ./inputnml/input_small.nml
    fields metadata: ./fields_metadata.yml
```

**Comment by frolovsa** [comment]
will do for both
------------------------------
Sergey Frolov (he/him)
PSL/OAR/NOAA

File: test/testinput/obslocalization_vertical.yml (hunk 1 above)
Comment (frolovsa) [reply test/testinput/obslocalization_vertical.yml:1]: @guillaumevernieres i fixed it up for the ctests in this PR. Still having difficulties fixing it for sqrt test. It complains about kmask.nc having wrong dimensions.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -602,6 +604,16 @@ soca_add_test( NAME obslocalization
                TEST_DEPENDS test_soca_gridgen
                NOCOMPARE )
 
+soca_add_test( NAME obslocalization_vertical
+               SRC  TestObsLocalization.cc
+               TEST_DEPENDS test_soca_gridgen
```
Comment (travissluka) [inline test/CMakeLists.txt]:
```suggestion
               TEST_DEPENDS test_soca_gridgen_small
```

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -602,6 +604,16 @@ soca_add_test( NAME obslocalization
...
+               NOCOMPARE )
+
+soca_add_test( NAME obslocalizations
+               SRC  TestObsLocalizations.cc
+               TEST_DEPENDS test_soca_gridgen
```
Comment (travissluka) [inline test/CMakeLists.txt]:
```suggestion
               TEST_DEPENDS test_soca_gridgen_small
```

**Review by travissluka — CHANGES_REQUESTED** [review]
one small correction in the cmakelists file
Thanks @frolovsa !

**Review by travissluka — APPROVED** [review]

**Comment by frolovsa** [comment]
@travissluka this should be ready to merge after the UFO PR is merged. 
https://github.com/JCSDA-internal/ufo/pull/1959

**Comment by rhoneyager** [comment]
The ufo PR is merged.

**Pull request merged**